tools/wheel_resolver/__init__.py

```diff
@@ -85,6 +95,10 @@ def main(
             click.echo(u)
             return
 
+    # We're currently hardcoding PyPI but we should consider allowing other
```

Comment:
nit: perhaps make this a TODO?

Reply:
Fair - just to share my thought process this is what I was thinking

**Assumptions**

- Most/all users of Python packages would download **at least some** packages from PyPI
- But some Please users might have some Python packages in **private** repositories (i.e. not PyPI) to complement the OSS packages that they get from PyPI (i.e. not as full replacements of PyPI)

This implies that the final solution would require the Wheel Resolver to locate packages from more than 1 package repository. If package names from PyPI and one of their private repos are equivalent the private repo's package takes precedence.

**Implementation**
In terms of implementation, maybe we could extend `python_wheel` to have the new `package_repository` attribute if we would be able to then pass that value to the Wheel Resolver? This would be the best solution in terms of both UX and technical complexity.

Alternatively, we could otherwise do it via `.plzconfig` by creating a new plugin as we do in this PR but the key downside is that we wouldn't know the package repo when we call the Wheel Resolver as it could be in one of multiple repos (PyPI or any of the potentially multiple private repos).

At a high-level, this would require the following changes:
- Allow `n` number of URLs of Python package repos to be set in `.plzconfig`, where the package repo precedence is given by the order in which they're listed
- PyPI's simple API is still hard-coded as the default option
- The Wheel Resolver would take the URLs as arguments, make API calls to all of them to find the package, and if a package with the same name is found in multiple repos download the one from the repo of highest precedence